File: src/main/java/view/GameView.java
package view;

import domain.Game;
import org.jetbrains.annotations.NotNull;

class GameView {
    private @NotNull TileBuffer mBuffer;
    private @NotNull BoardView  mBoardView;
    private @NotNull PlayerView mPlayerView;

    GameView(@NotNull TileBuffer buffer) {
        mBuffer     = buffer;
        mBoardView  = new BoardView(mBuffer);
        mPlayerView = new PlayerView(mBuffer);
    }

    void render(@NotNull Game game, Integer selectedPawnIndex) {
        mBoardView.render(game.getBoard());
        mPlayerView.render(game.getCurrentPlayer(), true, selectedPawnIndex);
        mPlayerView.render(game.getOtherPlayer(), true, selectedPawnIndex);
    }
}

File: Sources/CLI/view/GameView.swift
import Onitama

final class GameView {
  private let mBuffer: TileBuffer
  private let mBoardView: BoardView
  private let mPlayerView: PlayerView

  // -- lifetime --
  init(buffer: TileBuffer) {
    mBuffer = buffer
    mBoardView = BoardView(buffer: mBuffer)
    mPlayerView = PlayerView(buffer: mBuffer)
  }

  // -- commands --
  func render(game: Game, selectedPawnIndex: Int) {
    mBoardView.render(board: game.board)

    mPlayerView.render(
      player: game.currentPlayer,
      isCurrentPlayer: true,
      selectedPawnIndex: selectedPawnIndex
    )

    mPlayerView.render(
      player: game.otherPlayer,
      isCurrentPlayer: false,
      selectedPawnIndex: selectedPawnIndex
    )
  }
}
